File: frontend/src/components/Carousel/items/extrahop.js
/**
 * Author: Deric Le
 * Description: Handles ExtraHop API
 */

const FetchEH = async () => {
    try {
        const fetchCriticals = await fetch('http://localhost:8000/api/EH/detections/criticals');
        if (!fetchCriticals.ok) { throw new Error('Failed to fetch criticals from EH'); }

        const fetchHighs = await fetch('http://localhost:8000/api/EH/detections/highs');
        if (!fetchHighs.ok) { throw new Error('Failed to fetch criticals from EH'); }

        const criticals = await fetchCriticals.json();
        const highs = await fetchHighs.json();

        return {criticals: criticals, highs: highs};

    } catch (error) { 
        console.error('Error fetching data:', error);
        return null;
    }
    
};

export default FetchEH;

File: backend/capops_api/views/extrahop.py
"""
Author: Deric Le
Description: Define functions to be used for ExtraHop API
"""

import random 

from django.shortcuts import render
from rest_framework.decorators import api_view
from rest_framework.response import Response

@api_view(['GET'])
def getCriticals(request):
    criticals = random.randint(5, 15)
    return Response({'data': criticals})

@api_view(['GET'])
def getHighs(request):
    highs = random.randint(40, 100)
    return Response({'data': highs})